tests/test_clan_remaning_war_attacks.py

```diff
@@ -0,0 +1,131 @@
+
+import unittest
+from mockito import when
+from clash_royale_service import ClashRoyaleService
+from tests.resources import clash_royale_client_currentriverrace, clash_royale_client_responses
+
+
+class TestClanRemainingWarAttacks(unittest.TestCase):
+
+    def test_get_participated(self):
+        service = ClashRoyaleService()
+        participants = [
+            {"decksUsedToday": 3},
+            {"decksUsedToday": 4},
+            {"decksUsedToday": 0},
+            {"decksUsedToday": 1}
+        ]
+        assert service._get_participated(participants) == 3
+
+    def test_get_participated_no_participants(self):
+        service = ClashRoyaleService()
+        participants = [
+            {"decksUsedToday": 0},
+            {"decksUsedToday": 0},
+            {"decksUsedToday": 0},
+            {"decksUsedToday": 0}
+        ]
+        assert service._get_participated(participants) == 0
+
+    def test_get_players_remaining(self):
+        service = ClashRoyaleService()
+        participants = [
+            {"tag": "A", "decksUsedToday": 1},
+            {"tag": "B", "decksUsedToday": 2},
+            {"tag": "C", "decksUsedToday": 4},
+            # Suppose this player D left the clan
+            {"tag": "D", "decksUsedToday": 4}
+        ]
+        members_list = [
+            {"tag": "B"},
+            {"tag": "A"},
+            {"tag": "C"}
+        ]
+        assert service._get_players_remaining(participants, members_list) == 2
+    def test_get_decks_remaining(self):
+        service = ClashRoyaleService()
+        participants = [
+            {"tag": "A", "decksUsedToday": 1},
+            {"tag": "B", "decksUsedToday": 2},
+            {"tag": "C", "decksUsedToday": 4},
+            {"tag": "D", "decksUsedToday": 2}
+        ]
+        # Test 1 - All players are in clan and used some number of decks
+        members_list = [
+            {"tag": "B"},
+            {"tag": "A"},
+            {"tag": "C"},
+            {"tag": "D"}
+        ]
+        assert service._get_decks_remaining(participants, members_list) == 191
+        # Test 2 - Since D left, deckUsedToday for player D is 4 -> (200 - (1+2+4+4))
```

Comment:
Nice, I like it

Reply:
Might be worth having a completely separate test.... OR convert this test to a parameterised test and have a list of different test cases https://docs.pytest.org/en/6.2.x/example/parametrize.html 

Note: if you go with the latter approach you need to:
1. `import pytest`
2. Change `class TestClanRemainingWarAttacks(unittest.TestCase):` to `class TestClanRemainingWarAttacks(object):`